app/controllers/verify/finance_controller.rb

```diff
@@ -7,13 +7,18 @@ class FinanceController < ApplicationController
     before_action :confirm_step_allowed
     before_action :submit_idv_form, only: [:create]
     before_action :submit_idv_job, only: [:create]
+    before_action :refresh_if_not_ready, only: [:show]
```

Comment:
I find more controllers harder to follow, especially when the results are so tightly connected between actions like this.

I do agree about having fewer `before_actions` -- that was mostly in line with keeping the diff small.

I'll see what it looks like to extract a separate controller for this